### PR TITLE
docs(Dialog): MdDialogConfig ViewContainerRef description

### PR DESCRIPTION
**Pull request by M-a-c**
#3651 
I looked to find an example of what this is actually used for.
I tried to explain it a little bit more. (line 2)

https://github.com/angular/material2/blob/master/src/lib/core/portal/portal.ts#L79

Does this JsDoc summary suffice?

### CODE REVIEW

File: src/lib/dialog/dialog-config.ts [hunk 1]
```diff
@@ -15,6 +15,11 @@ export interface DialogPosition {
  * Configuration for opening a modal dialog with the MdDialog service.
  */
 export class MdDialogConfig {
+
+  /** Where the attached component should live in Angular's *logical* component tree.
+   * This is different from where the component *renders*, which is determined by
+   * the PortalHost (the component that was passed into MdDialog.open().
+   * Note the origin is necessary when the host is outside of the Angular application context. */
```
Comment (jelbourn) [inline src/lib/dialog/dialog-config.ts]: This isn't quite right. A little more accurate would be:
```ts
/** 
  * Where the attached component should live in Angular's *logical* component tree.
  * This affects what is available for injection and the change detection order for the 
  * component instantiated inside of the dialog. This does not affect where the dialog 
  * content will be rendered.
  */
```

Comment (M-a-c) [inline src/lib/dialog/dialog-config.ts]: Yeah this makes more sense. I'll switch that out.

File: src/lib/dialog/dialog-config.ts [hunk 2]
```diff
@@ -15,6 +15,11 @@ export interface DialogPosition {
  * Configuration for opening a modal dialog with the MdDialog service.
  */
 export class MdDialogConfig {
+
+  /** Where the attached component should live in Angular's *logical* component tree.
+  * This affects what is available for injection and the change detection order for the
+  * component instantiated inside of the dialog. This does not affect where the dialog
+  * content will be rendered. */
```
Comment (jelbourn) [inline src/lib/dialog/dialog-config.ts]: The JsDoc syntax isn't quite right. It goes
```ts
/**
 * Line one...
 * Line two...
 * Line three ...
 */
```

**Comment by M-a-c** [comment]
@jelbourn LGTY?
Sorry about the styling misunderstanding, I have looked through the style guide now. I apologize, I should have read that before contributing.

**Comment by M-a-c** [comment]
Noticed that my local and branches are all messed up. 
closing this creating new branch/w pr, using just these changes.

**Pull request closed without merging**